bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -578,6 +578,19 @@ impl Connection {
         }
     }
 
+    /// Adds a session ticket from a previous TLS connection to create a resumed session
+    pub fn set_session(&mut self, session: &[u8]) -> Result<&mut Self, Error> {
```

Comment:
Hmm. Is there maybe an opportunity for us to make our bindings a little more polite and have a real type here instead of the raw bytes?

My argument is something like this.
We have to use opaque structs in C because there isn't an easy way to control the visibility of struct members. But for rust we can easily control the visibility of struct members, so rather than having customers directly deal with the bytes for session tickets, we could wrap it in a higher level `SessionTicket` type, which could then be directly passed into `set_session`

Reply:
Trying to parse this. Are you saying that the output of the on_session_ticket cb should be used as an input for the set_session method?

TBH my brain has been so warped by C that I don't know what would be easier for users.

Reply:
I think all of our customers are going to have essentially copy-pasted code where they
1. save the data from the session
2. save the ticket age from the session
3. package that into a UserDefinedSessionType that can be stored (e.g. they want to store multiple tickets in a Vec)

And then when going to actually set the session
1. retrieve the UserDefinedSessionType
2. pull out the bytes
3. call `set_session`

Rather than making each of our customers solve that we have a 
```
pub struct RustBindingsSessionTicket {
    // members are not public, but the type is
    data: Vec<u8>
    lifetime: Duration
}
```

This would be the types that is used in the callback (which would pass ownership of the struct to the application, not a reference) and also the type that we'd use in the `set_session` method.

This is quite a bit more high level than our bindings have previously been, but I think it makes sense 🤷